extensions/default/src/MergeDataSource/index.js

```diff
@@ -0,0 +1,195 @@
+
+    const dataSourceDefs = Object.values(extensionManager.dataSourceDefs);
+    const promises = [];
+    let globalData = [];
+
+    for (const dataSourceDef of dataSourceDefs) {
+      const { configuration, sourceName } = dataSourceDef;
+      if (configuration && dataSourceNames.includes(sourceName)) {
+        const [dataSource] = extensionManager.getDataSources(sourceName);
+        const promise = executeFunction(dataSource, path, args);
+        promises.push(promise);
+
+        promise.then(data =>
+          processDedupedData({
+            sourceName,
+            globalData,
+            incomingData: data,
+            mergeKey,
+            level,
+            tagFunc,
+          })
+        );
+      }
+    }
+
+    await Promise.allSettled(promises);
+
+    return globalData.flat();
+  };
+
+  /**
+   * Calls a function with the given path and arguments on all data sources that are included in the given list of data source names.
+   * Returns an array of the results from each data source.
+   *
+   * @param {string} path - The path of the function to call on each data source.
+   * @param {Array} args - The arguments to pass to the function on each data source.
+   * @returns {Array} - An array of the results from each data source.
+   */
+  const callForAllDataSources = (path, args) => {
+    const dataSourceDefs = Object.values(extensionManager.dataSourceDefs);
+    const globalData = [];
+    for (const dataSourceDef of dataSourceDefs) {
+      const { configuration, sourceName } = dataSourceDef;
+      if (configuration && dataSourceNames.includes(sourceName)) {
+        const [dataSource] = extensionManager.getDataSources(sourceName);
+        const data = executeFunction(dataSource, path, args);
+        globalData.push(data);
+      }
+    }
+    return globalData.flat();
+  };
+
+  /**
+   * Calls the default data source with the given path and arguments.
+   * @param {string} path - The path to the function to be executed.
+   * @param {Array} args - The arguments to be passed to the function.
+   * @returns {*} - The result of executing the function.
+   */
+  const callForDefaultDataSource = (path, args) => {
+    const defaultDataSourceName = dataSourceNames[0];
+    const [dataSource] = extensionManager.getDataSources(defaultDataSourceName);
+    return executeFunction(dataSource, path, args);
+  };
+
+  /**
+   * Calls a method on the data source associated with the given display set's RetrieveAETitle or the default data source if none is found.
+   * @param {string} path - The path of the method to call.
+   * @param {Array} args - The arguments to pass to the method.
+   * @returns {*} - The result of calling the method.
+   */
+  const callByRetrieveAETitle = (path, args) => {
+    const [displaySet] = args;
+    const seriesMetadata = DicomMetadataStore.getSeries(
+      displaySet.StudyInstanceUID,
+      displaySet.SeriesInstanceUID
+    );
+    const defaultDataSourceName = dataSourceNames[0];
```

Comment:
Suggest adding a getter to get your default data source name based on the current circumstances.

Reply:
Or a const that you compute up front.

Reply:
Updated, thanks!